DNN Platform/Dnn.AuthServices.Jwt/Components/Common/Controllers/JwtController.cs

```diff
@@ -0,0 +1,489 @@
+﻿#region Copyright
+//
+// DotNetNuke® - http://www.dotnetnuke.com
+// Copyright (c) 2002-2016
+// by DotNetNuke Corporation
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+// documentation files (the "Software"), to deal in the Software without restriction, including without limitation
+// the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and
+// to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all copies or substantial portions
+// of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED
+// TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+// THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF
+// CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+// DEALINGS IN THE SOFTWARE.
+#endregion
+
+using System;
+using System.Collections.Generic;
+using System.IdentityModel.Tokens;
+using System.Linq;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Security.Claims;
+using System.Security.Cryptography;
+using System.Text;
+using Dnn.AuthServices.Jwt.Auth;
+using Dnn.AuthServices.Jwt.Components.Entity;
+using Dnn.AuthServices.Jwt.Data;
+using DotNetNuke.Entities.Portals;
+using DotNetNuke.Entities.Users;
+using DotNetNuke.Framework;
+using DotNetNuke.Instrumentation;
+using DotNetNuke.Security.Membership;
+using DotNetNuke.Web.Api;
+using Newtonsoft.Json;
+
+namespace Dnn.AuthServices.Jwt.Components.Common.Controllers
+{
+    internal class JwtController : ServiceLocator<IJwtController, JwtController>, IJwtController
+    {
+        #region constants, properties, etc.
+
+        private static readonly ILog Logger = LoggerSource.Instance.GetLogger(typeof(JwtController));
+        private static readonly HashAlgorithm Hasher = SHA384.Create();
+
+        private const int ClockSkew = 5; // in minutes; default for clock skew
+        private const int SessionTokenTtl = 60; // in minutes = 1 hour
+        private const int RenewalTokenTtl = 14; // in days = 2 weeks
```

Comment:
would it make sense for these times to be `TimeSpan` values instead of `int` values, so that their unit was more concretely connected to the value?